### PR TITLE
[SPARK-180|170|182] command parsing overhaul

### PR DESCRIPTION
**Pull request by theunkn0wn1**
This PR is a big one, it overhauls the way most commands are parsed.

Overarching idea is to parse the incoming commands using `pyparsing`, instead of raw string operations or regex.
By using `pyparsing` over regex, its far more readable.
By using `pyparsing` over direct string methods, its much more flexible *especially* for the more complicated commands.

This PR also has bug fixes for SPARK-170 and SPARK-182 as they fell into scope

### CODE REVIEW

File: src/commands/case_management.py [hunk 1]
```diff
@@ -175,32 +215,37 @@ async def cmd_case_management_clear(ctx: Context):
...
         await ctx.reply("No case with that name or number.")
         return
 
     async with ctx.bot.board.modify_rescue(rescue) as case:
-        case.client = ctx.words[2]
+        case.client = tokens.new_cmdr.strip()
         await ctx.reply(f"Client for {case.board_index} is now CMDR {case.client}")
 
 
+CODE_RED_PATTERN = suppress_first_word + rescue_identifier.setResultsName("subject")
```
Comment (Eearslya) [inline src/commands/case_management.py]: [Style] Should this not join its friends at the top of the file?

File: src/packages/ratmama/ratmama_parser.py [hunk 2]
```diff
@@ -126,49 +145,47 @@ async def handle_ratmama_announcement(ctx: Context) -> None:
...
+        client=client_name,
+        system=system_name,
+        irc_nickname=nickname,
+        code_red=not o2_status,
+        lang_id=lang_code,
+        platform=platform,
+    )
+    platform_signal = f"({rescue.platform.value.upper()}_SIGNAL)" if rescue.platform else ""
+    await ctx.reply(
+        f"DRILLSIGNAL - CMDR {rescue.client} - "
```
Comment (Eearslya) [inline src/packages/ratmama/ratmama_parser.py]: [Error] Hardcoded Drillsignal?

File: src/packages/ratmama/ratmama_parser.py [hunk 3]
```diff
@@ -232,21 +243,34 @@ async def handle_ratsignal(ctx: Context) -> None:
             platform = Platforms["XB"]
 
         elif "o2" in part.casefold():
-            code_red = "o2 ok" not in part.casefold()
+            code_red = "o2 ok" not in message.casefold()
 
         else:
             system = part
 
+    if system and "signal" in system.lower():
```
Comment (Eearslya) [inline src/packages/ratmama/ratmama_parser.py:250]: [Question] What case is this meant to catch?

File: tests/unit/test_ratmama.py [hunk 4]
```diff
@@ -152,8 +152,8 @@ async def test_announcer_reconnect_with_changes(bot_fx, async_callable_fx, monke
     await ratmama.handle_ratmama_announcement(context)
     await ratmama.handle_ratmama_announcement(context2)
 
-    assert async_callable_fx.was_called_with("System changed! Platform changed!"
-                                             " O2 Status changed, it is now CODE RED!")
+    assert async_callable_fx.was_called_with('Case #0 system, platform, O2 Status changed, '
+                                             'it is now CODE RED! O2 Status changed, it is now CODE RED!')
```
Comment (Eearslya) [inline tests/unit/test_ratmama.py]: [Question] Is it intentional that O2 status is announced twice?

File: src/packages/ratmama/ratmama_parser.py (hunk 3 above)
Comment (theunkn0wn1) [reply src/packages/ratmama/ratmama_parser.py:250]: I honestly don't recall what case this catches,  but it was a buggy edge case.

File: tests/unit/test_ratmama.py (hunk 4 above)
Comment (theunkn0wn1) [reply tests/unit/test_ratmama.py]: no, thats a bug :P

File: src/packages/ratmama/ratmama_parser.py (hunk 2 above)
Comment (theunkn0wn1) [reply src/packages/ratmama/ratmama_parser.py]: :tired_face:

File: src/commands/case_management.py (hunk 1 above)
Comment (theunkn0wn1) [reply src/commands/case_management.py]: probably. will fix

File: tests/unit/test_ratmama.py (hunk 4 above)
Comment (theunkn0wn1) [reply tests/unit/test_ratmama.py]: Resolved by fixing the erroneous underlying logic.

File: src/packages/ratmama/ratmama_parser.py (hunk 2 above)
Comment (theunkn0wn1) [reply src/packages/ratmama/ratmama_parser.py]: Resolved hardcoded signal.

File: src/commands/case_management.py (hunk 1 above)
Comment (theunkn0wn1) [reply src/commands/case_management.py]: Moved.

**Comment by theunkn0wn1** [comment]
I removed two of the hypothesis tests that frequently failed health checks.
It was a good idea to have them, but more effort needs to be made in stabilizing them.

**Review by Eearslya — APPROVED** [review]
